tests/Validation/ValidationEmailRuleTest.php

```diff
@@ -80,11 +81,12 @@ protected function assertValidationRules($rule, $values, $expectToPass, $expecte
                 $customValidationMessage ? [self::ATTRIBUTE.'.email' => $customValidationMessage] : []
             );
 
-            $this->assertSame($expectToPass, $v->passes());
+            $this->assertSame($expectToPass, $v->passes(), 'Expected email input '.$value.' to '.($expectToPass ? 'pass' : 'fail').'.');
```

Comment:
You might want to use a [`#[DataProvider]`](https://docs.phpunit.de/en/10.5/attributes.html#data-provider) for this

Reply:
Good point, updated to `#[TestWith]`

Reply:
Nice, never used it but looks neat 👍🏻